samples/net/coap_client/src/main.c

```diff
@@ -250,74 +94,71 @@ static void wait_for_network(void)
 	k_mutex_unlock(&network_connected_lock);
 }
 
-/**
- * @brief Infinite CoAP request loop
- */
-static void coap_loop(void)
+/* Callback to handle the response */
+void response_cb(int16_t code, size_t offset, const uint8_t *payload,
+				size_t len, bool last_block, void *user_data)
+{
+	if (code >= 0) {
+		LOG_INF("CoAP response: code: 0x%x, payload: %s", code, payload);
+	} else {
+		LOG_INF("Response received with error code: %d", code);
+	}
+}
+
+/* Create the request */
+struct coap_client_request req = {
```

Comment:
IMO it will look a bit more tidy to move declaration of this structure to the top with the other variables, and make it static.